libs/request.js

```diff
@@ -54,38 +54,31 @@ module.exports = function (options = {}) {
 
   if (options.debug) requestdebug(request)
 
-  const requestOptions = {}
-
-  requestOptions.json = options.json
-  requestOptions.jar = options.jar
-  requestOptions.headers = options.headers
-  requestOptions.followAllRedirects = options.followAllRedirects
-  requestOptions.strictSSL = options.strictSSL
-
   if (options.cheerio) {
-    requestOptions.transform = function (body, response, resolveWithFullResponse) {
```

Comment:
The problem with this change is that the user cannot change resolveWithFullResponse just for one call to request. Example wich I meet a lot in connectors: 

```javascript
const {request} = require('cozy-konnector-libs')
const rq = request({
  cheerio: true,
  json: false
})

rq('https://test.url')
.then($ => {
  const token = $('.token').val()
  return rq({
    resolveWithFullResponse: true
  })
})
.then(response => {
  // test response.statusCode and get cheerio instance in response.body
})
```

To me, the request function in cozy-konnector-libs was just a `request.defaults` shortcut with better defaults for connectors. But maybe that is not clear.

Reply:
Yes, actually it confused me a lot. IMHO if we provide an instance of `request` we should encapsulate it in another function or object than one called also `request` or `rq`. Also, I am not sure that set a custom config of `request` for every konnector is a good pattern. I was lost when I looked at the `request-promise` documentation because the konnector lib behaviour did not match the documented API. Maybe it's a litte too "magical" to me.

Reply:
> To me, the request function in cozy-konnector-libs was just a request.defaults shortcut with better defaults for connectors. But maybe that is not clear.

Indeed it is a bit tricky to have a function named exactly like another one and having a different behavior.

> Maybe it's a litte too "magical" to me.

I agree.